convert-sg-db.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python
+
+import json
+import commonmark
+from datetime import date
+
+
+def create_biblio(db):
+    biblio = {}
+
+    for workstream in db["workstreams"]:
+        for standard in workstream["standards"]:
+            reference, contents = standard_or_idea_to_biblio(standard)
+            contents["title"] += " Standard"
+            biblio[reference] = contents
+
+    for idea in db["ideas"]:
+        reference, contents = standard_or_idea_to_biblio(idea)
+        contents["status"] = "Living Idea"
+        biblio[reference] = contents
+
+    for biblio_item in db["biblio"]:
+        reference, contents = biblio_item_to_biblio(biblio_item)
+        biblio[reference] = contents
+
+    string = json.dumps(biblio, ensure_ascii=False, allow_nan=False, sort_keys=True, indent=2)
+    return string + "\n"
+
+def standard_or_idea_to_biblio(document):
+    return (document["reference"], {
+        "authors": [author_to_biblio(author) for author in document["authors"]],
+        "href": document["href"],
+        "title": document["name"]
+    })
+
+def biblio_item_to_biblio(item):
+    contents = {
+        "href": item["href"],
+        "title": item["title"]
+    }
+    if len(item["authors"]):
+        contents["authors"] = [author_to_biblio(author) for author in item["authors"]]
+    if "obsoletedBy" in item:
+        contents["obsoletedBy"] = item["obsoletedBy"]
+    return (item["reference"], contents)
+
+def author_to_biblio(person):
+    return person["name"]
+
+
+def create_spec_whatwg_org(db, template):
+    content = "<p>The WHATWG works on a number of technologies that are fundamental parts of the web platform. They are organised somewhat arbitrarily based on the preferences of those editing the standard for those technologies.</p>\n\n"
+    content += "<dl>"
+    for workstream in db["workstreams"]:
+        for standard in workstream["standards"]:
+            content += standard_or_idea_to_html(standard)
+    content += "</dl>\n\n<p>The WHATWG also works on a <a href=\"https://idea.whatwg.org/\">number of ideas</a> that aspire to become standards one day.</p>\n\n"
+
+    template = template.replace("<a href=\"https://spec.whatwg.org/\">Standards</a>", "<a>Standards</a>")
+    return wrap_in_site_template(template, "Standards", content)
+
+def standard_or_idea_to_html(document):
+    output = "\n <dt><a href=\"{0}\">{1}</a></dt>".format(document["href"], document["name"])
+    return output + "\n <dd>{}</dd>\n".format(commonmark.commonmark(document["description"])[:-1]) # Strip trailing \n
+
+def wrap_in_site_template(template, title, content):
```

Comment:
https://github.com/whatwg/whatwg.org/commit/4bcc3c3e4b5c5abc176ef447f96e7f12c5cb615c reveals a problem in idea.whatwg.org/index.html, where the link will be broken because site-template.html assumes the result will be on whatwg.org and links /.

Reply:
I addressed this by always using the full link. This is a bit redundant for some URLs, but does work kind of nice if you end up hosting the same content on another URL for preview reasons. 😊

Reply:
Yep, that works!